Source/Particles/Collision/BinaryCollision/PhotonPhoton/SinglePhotonPhotonCollisionEvent.H

```diff
@@ -0,0 +1,183 @@
+                               amrex::ParticleReal w1, amrex::ParticleReal w2,
+                               const amrex::Real& dt, const amrex::ParticleReal& dV, const int& pair_index,
+                               index_type* AMREX_RESTRICT p_mask,
+                               amrex::ParticleReal* AMREX_RESTRICT p_pair_reaction_weight,
+                               const amrex::ParticleReal& event_multiplier,
+                               const int& multiplier_ratio,
+                               const amrex::ParticleReal& probability_threshold,
+                               const amrex::ParticleReal& probability_target_value,
+                               const amrex::RandomEngine& engine)
+{
+    // General notations in this function:
+    //     x_sq denotes the square of x
+    //     x_star denotes the value of x in the center of mass frame
+
+    using namespace amrex::literals;
+
+    const amrex::ParticleReal w_min = amrex::min(w1, w2);
+    const amrex::ParticleReal w_max = amrex::max(w1, w2);
+
+    constexpr auto one_pr = amrex::ParticleReal(1.);
+    constexpr auto two_pr = amrex::ParticleReal(2.);
+    constexpr auto one_half_pr = amrex::ParticleReal(1./2.);
+    constexpr amrex::ParticleReal c = PhysConst::c;
+    constexpr amrex::ParticleReal c_sq = c*c;
+    constexpr amrex::ParticleReal me = PhysConst::m_e;
+    auto constexpr pow2 = [](double const x) { return x*x; };
+
+    // Compute momenta in the lab frame
+    const amrex::ParticleReal p1x = u1x * me;
+    const amrex::ParticleReal p1y = u1y * me;
+    const amrex::ParticleReal p1z = u1z * me;
+    const amrex::ParticleReal p2x = u2x * me;
+    const amrex::ParticleReal p2y = u2y * me;
+    const amrex::ParticleReal p2z = u2z * me;
+    const amrex::ParticleReal p1 = std::sqrt(pow2(p1x)+pow2(p1y)+pow2(p1z));
+    const amrex::ParticleReal p2 = std::sqrt(pow2(p2x)+pow2(p2y)+pow2(p2z));
+
+    // Compute Lorentz factor gamma in the lab frame
+    const amrex::ParticleReal g1 = p1 / ( me * c );
+    const amrex::ParticleReal g2 = p2 / ( me * c );
+
+    // Compute the cosine of the angle between the two photon momenta in the lab frame
+    const amrex::ParticleReal cos_ang = (p1x*p2x+p1y*p2y+p1z*p2z)/(p1*p2);
+
+    // Energy squared of each and every particle involved in the collision
+    // (both colliding photons and the produced electron and positron)
+    // in the center of momentum frame (they are all equal),
+    // calculated using the Lorentz invariance of the total four-momentum norm
+    const amrex::ParticleReal E_star_sq = one_half_pr*c_sq*p1*p2*(one_pr - cos_ang);
+
+    // Kinetic energy of each particle in the center of momentum frame
+    const amrex::ParticleReal E_star = std::sqrt(E_star_sq);
+
+    // Compute linear Breit Wheeler cross section as a function of the kinetic energy
+    // in the center of momentum frame
+    auto lbw_cross_section = amrex::ParticleReal(0.);
+    lbw_cross_section = LinearBreitWheelerCrossSection(E_star);
```

Comment:
Would it be possible to do
```suggestion
    const auto lbw_cross_section = LinearBreitWheelerCrossSection(E_star);
```
?